### PR TITLE
Fancy FS led startup sequence

### PR DESCRIPTION
**Pull request by Batambatam**
Summary of changes:

Just a fancy startup Leds sequence, using function switches led.
Only for TPro.

### CODE REVIEW

**Comment by pfeerick** [comment]
That looks really nice. I suppose you didn't add it to shutdown animation also because of the strobing (because FS led state is still being managed by evalFunctionSwitches? If you're interested in add that also... this should work... feel free to add or improve ;)

```diff
diff --git a/radio/src/gui/128x64/startup_shutdown.cpp b/radio/src/gui/128x64/startup_shutdown.cpp
index 6afb2b490..0e87d4301 100644
--- a/radio/src/gui/128x64/startup_shutdown.cpp
+++ b/radio/src/gui/128x64/startup_shutdown.cpp
@@ -67,6 +67,16 @@ void drawShutdownAnimation(uint32_t duration, uint32_t totalDuration, const char
   lcdRefreshWait();
   lcdClear();
 
+#if defined(RADIO_TPRO)
+  uint8_t index2 = limit<uint8_t>(0, duration / (totalDuration / 7), 6);
+
+  for (uint8_t j = 0; j < 6; j++) {
+    fsLedOff(j);
+    if (6 - index2 > j)
+      fsLedOn(j);
+  }
+ #endif
+
   for (uint8_t i = 0; i < 4; i++) {
     if (4 - index > i) {
       lcdDrawFilledRect(LCD_W / 2 - 18 + 10 * i, LCD_H / 2 - 3, 6, 6, SOLID, 0);
diff --git a/radio/src/switches.cpp b/radio/src/switches.cpp
index 736938b8a..6c2f6c0d1 100644
--- a/radio/src/switches.cpp
+++ b/radio/src/switches.cpp
@@ -145,10 +145,12 @@ void evalFunctionSwitches()
       storageDirty(EE_MODEL);
     }
 
+if (!pwrPressed()) {
     if (getFSLogicalState(i))
       fsLedOn(i);
     else
       fsLedOff(i);
+}
   }
 }
 #endif

```

File: radio/src/gui/128x64/startup_shutdown.cpp [hunk 1]
```diff
@@ -35,10 +35,18 @@ void drawStartupAnimation(uint32_t duration, uint32_t totalDuration)
     return;
 
   uint8_t index = limit<uint8_t>(0, duration / (totalDuration / 5), 4);
-
   lcdRefreshWait();
   lcdClear();
 
+#if defined(RADIO_TPRO)
```
Comment (raphaelcoeffic) [inline radio/src/gui/128x64/startup_shutdown.cpp:41]: This should be using the proper define for function switches rather than TPRO.

Comment (pfeerick) [reply radio/src/gui/128x64/startup_shutdown.cpp:41]: In that vein, `NUM_FUNCTIONS_SWITCHES` gets rid of the magic number "6" also ;)

**Comment by pfeerick** [comment]
Nah, just means I've been poking around in the code too much. For future reference, I was in luck... I just searched for usage of fsOn /fsOff and that was indeed it.

**Review by pfeerick — APPROVED** [review]

**Pull request merged**